pkg/system/phase2_creating.go

```diff
@@ -746,6 +746,8 @@ func (r *Reconciler) ReconcileRootSecret() error {
 func (r *Reconciler) ReconcileDB() error {
 	var err error
 	if r.NooBaa.Spec.DBType == "postgres" {
+		r.PostgresDBConf.Namespace = r.NooBaaPostgresDB.Namespace
+		util.KubeCreateSkipExisting(r.PostgresDBConf)
```

Comment:
what about error checking? check out other examples using KubeCreateSkipExisting

Reply:
is it better?

https://github.com/baum/noobaa-operator/blob/60bf90429750c74a8e8f85568b607116c03edc01/pkg/system/phase2_creating.go#L751-L758

Reply:
also moved Postgres configMap Namespace assignment to the reconciler constructor:

https://github.com/noobaa/noobaa-operator/compare/0fcb889..3c3fe2f

Reply:
That would work but I think maybe it's worth to split util.KubeCreateSkipExisting into two a bit different functions:
- util.KubeCreateSkipExisting - almost like KubeCreateSkipExisting is today, only change is if already exists then return true (instead of false)
- util.KubeCreateFailExisting - this will do exactly what KubeCreateSkipExisting does today (!) yes the name is simply wrong because it returns false when already exists, just like with any other error.

This will make more sense for cases like yours where the we want to create it once and if it exists that's ok:
```golang
if !util.KubeCreateSkipExisting(r.PostgresDBConf) {
    return fmt.Errorf(...)
}
```

And for all the current cases that use KubeCreateSkipExisting we can choose which option fits better.
In most CLI calls it doesn't really matter because no body checked what the function returns.
But for cases that did check the return value, we want to see what was the poets intention...
WDYT?

Reply:
why not use KubeCreateOptional? the optional section is on a missing CRD

Reply:
because that's not the same case.
KubeCreateOptional is for CRDs that are optional in the cluster.
In these cases we want an error checking for the existence of the CR itself.

Reply:
@guymguym sounds good. renamed symbol KubeCreateSkipExisting -> KubeCreateFailExisting and extracted kubeCreateSkipOrFailExisting() and implemented both skip and fail paths in terms of this new extracted function.

See https://github.com/baum/noobaa-operator/blob/39c5a4ea3cad97d10aef872aa3c86ca68cbc6966/pkg/util/util.go#L206-L263

Now the flow of new configMap creation looks solid 😀:

https://github.com/baum/noobaa-operator/blob/39c5a4ea3cad97d10aef872aa3c86ca68cbc6966/pkg/system/phase2_creating.go#L749-L755